src/Disks/ObjectStorages/S3/S3ObjectStorage.cpp

```diff
@@ -255,7 +257,7 @@ void S3ObjectStorage::removeObjects(const std::vector<std::string> & paths)
     if (!s3_capabilities.support_batch_delete)
     {
         for (const auto & path : paths)
-            removeObject(path);
+            removeObjectImpl(path, if_exists);
```

Comment:
GitHub somehow won't let me add a comment on an unchanged line and it's kindof unrelated, but the scope of the variables in l. 254/255 could be decreased.